polaris/hub/polarisfs.py

```diff
@@ -202,14 +223,7 @@ def pipe_file(
         hub_response_body = response.json()
         signed_url = hub_response_body["url"]
 
-        sha256_hash = hashlib.sha256(content).hexdigest()
-
-        headers = {
-            "Content-Type": "application/octet-stream",
-            "x-amz-content-sha256": sha256_hash,
-            "x-amz-date": datetime.now(timezone.utc).strftime("%Y%m%dT%H%M%SZ"),
-            **hub_response_body["headers"],
-        }
+        headers = {"Content-Type": "application/octet-stream", **hub_response_body["headers"]}
```

Comment:
**Suggestion**
Cloudflare R2 [supports](https://developers.cloudflare.com/r2/api/s3/api/#implemented-object-level-operations) the `Content-MD5` HTTP header on PUT operations. You could add it here to ensure integrity on the fragment being uploaded here.